### PR TITLE
Google analytics: use GA4 site tag

### PR DESCRIPTION
**Pull request by chalin**
- Contributes to #35299

Preparatory step:

- [x] Add the GA4 ID via Netlify config. [Done by @onlydole, thanks!]

### CODE REVIEW

**Comment by chalin** [comment]
I'm going to put this in draft status until we get some resolution of:

- #36011

**Comment by chalin** [comment]
@jimangel @sftim et all: this is ready for review.

File: layouts/partials/head.html [hunk 1]
```diff
@@ -36,7 +25,7 @@
 {{- template "_internal/schema.html" . -}}
 {{- template "_internal/twitter_cards.html" . -}}
 {{- if hugo.IsProduction -}}
-{{ template "_internal/google_analytics_async.html" . }}
+{{ template "_internal/google_analytics.html" . }}
```
Comment (sftim) [inline layouts/partials/head.html:28]: Why this change? Is that adopted from upstream Docsy?

**Review by sftim — COMMENTED** [review]
For `config.toml` we currently have
> ```
> [services]
> [services.googleAnalytics]
> id = "G-JPP6RFM2BP"
> ```

I would keep that there but use a fake ID and comment it out _with an explanation_ because folks may well look at the repo and wonder how it all works.

File: layouts/partials/head.html (hunk 1 above)
Comment (chalin) [reply layouts/partials/head.html:28]: > Why this change?

From [Use the Google Analytics Template](https://gohugo.io/templates/internal/#use-the-google-analytics-template) in the Hugo docs:

> When using Google Analytics v4 use `_internal/google_analytics.html`.

As for Docsy: the logic is different from the local change made in `head.html`.

**Comment by chalin** [comment]
@sftim - I've documented the fake ID and mentioned where the real one is. PTAL

**Comment by chalin** [comment]
/hold

Hold until we can set the GA ID via Netlify (as is mentioned in the opening comment). For details, see https://github.com/kubernetes/website/issues/36011#issuecomment-1218093743.

**Comment by kbhawkey** [comment]
> /hold
> 
> Hold until we can set the GA ID via Netlify (as is mentioned in the opening comment). For details, see [#36011 (comment)](https://github.com/kubernetes/website/issues/36011#issuecomment-1218093743).

Ah, okay
/lgtm cancel

**Comment by chalin** [comment]
@kbhawkey - I think that your LGTM was fine, independent of the hold on the PR.

**Comment by chalin** [comment]
/unhold

Thanks to help from @onlydole, this is now ready to be merged (and LGTM'd again @kbhawkey :)).

**Review by onlydole — APPROVED** [review]
Thank you, @chalin and @sftim! This looks good to me and I approve!

**Comment by reylejano** [comment]
Thank you for addressing comments and suggestions 
/lgtm

**Comment by chalin** [comment]
Thanks @reylejano!

Questions: When will this appear on the production server? Where is it deployed to in the meantime?

**Comment by chalin** [comment]
Scratch those questions @reylejano, I see that this is live, and all is running well!
